src/spglib.c

```diff
@@ -117,8 +117,8 @@ static int get_symmetry_from_dataset(
     const int num_atom, const double symprec, const double angle_tolerance);
 static MagneticSymmetry *get_symmetry_with_site_tensors(
     int equivalent_atoms[], int **permutations, double primitive_lattice[3][3],
-    const int max_size, const Cell *cell, const int with_time_reversal,
-    const int is_axial, const double symprec, const double angle_tolerance,
+    const Cell *cell, const int with_time_reversal, const int is_axial,
+    const double symprec, const double angle_tolerance,
```

Comment:
These I presume are resolving the warnings displayed. It might be difficult to navigate the git history if we squash this PR (confusion between which part enabled it and which one resolved the warnings). Make sure you do at least one of:
- interactively rebase the commit history before merging to have the final commit history we want im the end
- separate parts of the PR logically, e.g. enabling the warnings and resolving them (as long as it doesn't affect the CI)